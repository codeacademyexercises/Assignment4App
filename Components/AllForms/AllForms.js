/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import React, { Component } from 'react';
import {
  Platform, View, SafeAreaView, ScrollView, TouchableHighlight,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import PropTypes from 'prop-types';
import styles from './AllForms.style';
import Header from '../Header/Header';
import FormPortal from '../FormPortal/FormPortal';

const axios = require('axios');

export default class App extends Component {
  static navigationOptions = {
    header: null,
  }

  state = {
    FormDetails: [],
  }

  getFormData = async (url) => {
    await axios.get(url).then((data) => { this.setState({ FormDetails: data.data }); });
  }

  async componentDidMount() {
    if (Platform.OS === 'android') {
      console.log('android');
      await this.getFormData('http://10.0.2.2:3005/displayforms');
    } else {
      await this.getFormData('http://localhost:3005/displayforms');
    }
    const { FormDetails } = this.state;
    console.log(FormDetails, 'hi');
  }

  // getFormData = async (url) => {
  //   await axios.get(url).then((data) => { this.setState({ FormDetails: data.data }); });
  // }

  render() {
    const Forms = [];
    const { FormDetails } = this.state;
    const { navigation } = this.props;
    console.log(FormDetails);
    FormDetails.forEach(Form => Forms.push(<FormPortal FormName={Form.FormName} createdOn={Form.createdAt} />));
    return (
      <SafeAreaView style={styles.container}>
        <Header />
        <ScrollView style={styles.FormsScrollView}>
          <View style={styles.forms}>
            {Forms}
          </View>
        </ScrollView>
        <TouchableHighlight style={styles.createFormButton} onPress={() => { navigation.navigate('NewForm'); }}>
          <Icon name="plus" style={styles.icon} />
        </TouchableHighlight>
      </SafeAreaView>
    );
  }
}

App.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};
