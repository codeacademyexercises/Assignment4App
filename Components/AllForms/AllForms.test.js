import React from 'react';
import renderer from 'react-test-renderer';
import * as axios from "axios";
import AllForms from './AllForms';
import { shallow,configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockAxios from 'axios';

configure({ adapter: new Adapter() });

jest.mock("axios");

const result=[{
    FormFields: "FirstName,LastName",
    FormName: "Personal Details",
    NumberOfFields: 2,
    createdAt: "2019-02-27T06:47:39.504Z",
    id: 2,
    updatedAt: "2019-02-27T06:47:39.504Z",},
    {
    FormFields: "Date,rating,feedback",
    FormName: "CodeAcademy Feedback",
    NumberOfFields: 3,
    createdAt: "2019-02-27T06:54:59.297Z",
    id: 3,
    updatedAt: "2019-02-27T06:54:59.297Z",
    },
]
describe('<AllForms /> ',()=>{
    it('snapshot test',()=>{
        axios.get.mockImplementation( () => Promise.resolve({ data: result }) );
        const tree = renderer.create(<AllForms />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('expect getFormData() to call axios',async ()=>{
        mockAxios.get.mockImplementationOnce(() =>
            Promise.resolve({
            data: result,
            })
        );
        const wrapper = shallow(<AllForms />);
        const instance = wrapper.instance();
        await instance.getFormData('');
        expect(wrapper.state('FormDetails')).toEqual(result);
    });
    it('testing navigation is called when create new form buttom is clicked',()=>{
        const navigation = { navigate: jest.fn() };
        const wrapper = shallow(<AllForms navigation={navigation} />);
        const instance = wrapper.instance();
        wrapper.find('TouchableHighlight').simulate('press');
        expect(wrapper.prop('navigation').navigate).toHaveBeenCalled();
    })
})