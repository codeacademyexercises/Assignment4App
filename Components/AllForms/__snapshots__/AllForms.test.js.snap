// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllForms />  snapshot test 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#DAE5F0",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#4154AF",
        "flexDirection": "row",
        "height": 100,
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
        }
      }
    >
      Awesome Forms
    </Text>
  </View>
  <RCTScrollView
    style={Object {}}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "space-around",
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "backgroundColor": "#4154AF",
        "borderRadius": 60,
        "height": 50,
        "justifyContent": "center",
        "marginRight": 30,
        "width": 50,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "normal",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ÔÅß
    </Text>
  </View>
</RCTSafeAreaView>
`;
